chat-ui-manager-app/src/routes/+page.svelte

```diff
@@ -0,0 +1,141 @@
+<script lang="ts">
+	import { onMount } from 'svelte';
+	import { MachineStatus } from '$lib/types';
+	import { env } from '$env/dynamic/public';
+
+	const chatUIUrl = env.PUBLIC_CHAT_UI_URL;
+	const chatUIDelayMs = Number(env.PUBLIC_CHAT_UI_DELAY_SEC) * 1000 || 30_000; // Default to 30 seconds if not set
+
+	let machineStatus: MachineStatus = $state(MachineStatus.UNKNOWN);
+	let chatUIAvailable = $state(false);
+
+	onMount(() => {
+		const controller = new AbortController();
+		checkMachineStatus(controller.signal);
+
+		// Cleanup function to abort the fetch request when the component is destroyed
+		return () => {
+			controller.abort();
+			console.log('Fetch aborted');
+		};
+	});
+
+	async function checkMachineStatus(signal?: AbortSignal) {
+		try {
+			const res = await fetch('/api/vm/status', { signal });
+			if (!res.ok) throw new Error('Failed to get status');
+			const { status, startedAt } = await res.json();
+			machineStatus = status; // E.g. MachineStatus.RUNNING or MachineStatus.STOPPED etc.
+
+			if (startedAt) {
+				const startTime = new Date(startedAt).getTime();
+				const elapsedTime = Date.now() - startTime;
+				// If the machine is running longer than the delay, assume chat UI is available
+				if (elapsedTime >= chatUIDelayMs) {
+					chatUIAvailable = true;
+				} else {
+					const remainingTime = chatUIDelayMs - elapsedTime;
+					setChatUIDelay(remainingTime);
+				}
+			}
+		} catch (err) {
+			if (err instanceof Error && err.name === 'AbortError') {
+				console.error('Fetch aborted');
+			} else {
+				console.error(err);
+				machineStatus = MachineStatus.UNKNOWN; // Revert if it fails
+			}
+		}
+	}
+
+	async function startMachine() {
+		machineStatus = MachineStatus.STARTING;
+		try {
+			const res = await fetch('/api/vm/start', { method: 'POST' });
+			if (!res.ok) throw new Error('Failed to start machine');
+			const data = await res.json();
+			machineStatus = data.status; // E.g. MachineStatus.RUNNING
+
+			if (machineStatus === MachineStatus.RUNNING) {
+				setChatUIDelay(chatUIDelayMs); // Once the machine is running, set the delay
+			}
+		} catch (err) {
+			console.error(err);
+			machineStatus = MachineStatus.STOPPED; // Revert if it fails
+		}
+	}
+
+	async function stopMachine() {
+		machineStatus = MachineStatus.STOPPING;
+		chatUIAvailable = false; // Reset availability of the Chat-UI page when stopping the VM
+		try {
+			const res = await fetch('/api/vm/stop', { method: 'POST' });
+			if (!res.ok) throw new Error('Failed to stop machine');
+			const data = await res.json();
+			machineStatus = data.status; // E.g. MachineStatus.STOPPED
+		} catch (err) {
+			console.error(err);
+			machineStatus = MachineStatus.RUNNING; // Revert if it fails
```

Comment:
You might want to set the available status back to `true` here if it was `true` before